diff --git a/src/zeta14cubicmult.f90 b/src/zeta14cubicmult.f90
index e0bc00d..655cd1e 100644
--- a/src/zeta14cubicmult.f90
+++ b/src/zeta14cubicmult.f90
@@ -868,6 +868,7 @@ program compute_hardy_fast
 
   !     All Done
 
+  call MPI_Finalize(IERR)
   STOP
 
 
@@ -899,8 +900,6 @@ program compute_hardy_fast
 4008 FORMAT(A12,E47.40)
 
   
-  call MPI_Finalize(IERR)
-
 end program compute_hardy_fast
 
 
